lib/deck.rb

```diff
@@ -0,0 +1,34 @@
+class Deck
+  attr_accessor :cards
+
+  def initialize(cards)
+    @cards = cards
+  end
+
+  def rank_of_card_at(index)
+    @cards[index].rank
+  end
+
+  def high_ranking_cards
+    @cards.select do |card|
+      card.rank >= 11
+    end
+  end
+
+  def percent_high_ranking
+    last_digit = (high_ranking_cards.length / @cards.length.to_f * 100).to_s.slice(-1).to_i
+    if last_digit >= 5
+      (high_ranking_cards.length / @cards.length.to_f * 100).ceil(2)
+    else
+      (high_ranking_cards.length / @cards.length.to_f * 100).floor(2)
+    end
```

Comment:
I think this is a little overcomplicated. The following should also solve the problem and is a lot more readable:
`(high_ranking_cards.count / cards.count.to_f * 100).round(2)`